module-2/backend/week_two.md

```diff
@@ -1,45 +1,126 @@
-6. Define foreign key, primary key, and schema.
-7. Describe the relationship between a foreign key on one table and a primary key on another table.
 8. What are the parts of an HTTP response?
 
+  Response code, message that corresponds to the code, header, and body.
 
 ### Optional Questions
 
-1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
-2. Name your three favorite ActiveRecord rake tasks and describe what they do.
-3. What two columns does `t.timestamps null: false` create in our database?
-4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.  
+
+  order: this method orders the data according to the parameters you specify (can order by a specific column. it defaults to ascending order, so include DESC in the argument if you want it sorted in descending order.
+
+  select: this method allows you to determine which information you want to include in a query; you can select all information from a table by writing tablename.*
+
+  where: this method selects all resources that meet the query criteria; I like that this method allows you to set certain max/min values and then select only the resources that fall in the criteria.
+
+  maximum: this method returns the value of the maximum value in the column that's passed in as an argument;
+
+  pluck: this method returns values in a column from a collection, and these values are returned as an array.
+
+2. Name your three favorite ActiveRecord rake tasks and describe what they do.  
+
+  rake db:test:prepare updates the test database to include the most up to date schema.
+
+  rake db:create_migration allows you to create a migration.
+
+  rake db:migrate runs your migrations to update the schema and the database.
+
+3. What two columns does `t.timestamps null: false` create in our database?  
+
+  created_at and updated_at
+
+4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?  
+
+  This is a one to many relationship because a school has many teachers but teachers only belong to a single school (unless we're talking about social workers/nurses/coaches who split time between multiple schools).
+
+  The teachers table will have a foreign key of school_id in order to relate these two separate resources.
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+  I would have to add a foreign key of school id to the teacher table, add ```belongs_to :school``` in the teacher.rb model and add ```has_many :teachers``` in the school.rb model.
+
 6. Give an example of when you might want to store information besides ids on a join table.
-7. Describe and diagram the relationship between patients and doctors.
+
```

Comment:
maybe a grade on a `student_classes` join table associated with just that class for that student?